sql-plugin/src/main/scala/com/nvidia/spark/rapids/DateUtils.scala

```diff
@@ -92,6 +104,17 @@ object DateUtils {
     )
   }
 
+  def fetchSpecialDates(unit: DType): Map[String, Scalar] = unit match {
+    case DType.TIMESTAMP_DAYS =>
+      DateUtils.specialDatesDays.map { case (k, v) => k -> Scalar.timestampDaysFromInt(v) }
```

Comment:
nit: There is a period of time from when this function is called and scalars are populated into the map and when those scalars are setup to be closed.  I personally would prefer to see this return a `Map[String, () -> Scalar]` so that the scalar is generated as it is used, and then closed as soon as it is done.